### ISSUE TITLE
Add way to have ResearchItems that autocompletes once all parents are unlocked

### ISSUE THREAD

**Issue by MalkContent** [Azanor/thaumcraft-5-api#23]
Say we got `ResearchItem("MYVIRTUALITEM", "MYCATEGORY").setParents("FOCUSPOUCH", "ELEMENTALTOOLS")` and I want `"MYVIRTUALITEM"` to autounlock once both `"FOCUSPOUCH"` and `"ELEMENTALTOOLS"` are complete.
As far as I can tell this isn't possible at the moment. I tried .setAutoUnlock(), but that just unlocks it right away, regardless of parents. I tried adding nothing (in hopes that virtual ResearchItems get handled differently in that regard), but that did... nothing x). I tried getting both parent researches and adding `"MYVIRTUALITEM"` to their sibling list, which sort of worked. The problem there was that researching `"FOCUSPOUCH"` would directly finish `"MYVIRTUALITEM"`, and presumably because finishing a research backchecks that all parents are unlocked, also finish `"ELEMENTALTOOLS"` and all its parents.
Bottom line: I'd like the last method to work as I intented it. Maybe add some functions to make setting up this kind unlocking easier.
If it still makes sense to contribute to this Api, what with TC II in the air, please tell me and I can see if I can figure out how to make a proper pull request :D
